### PR TITLE
[VEN-105] Product states - pre-Orders support - drop available_on logic

### PR DESCRIPTION
**Pull request by kgorazd**

### CODE REVIEW

File: app/helpers/spree/admin/products_helper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Spree
+  module Admin
+    module ProductsHelper
+      # will return a human readable string
+      def available_status(product)
+        return Spree.t(:archived) if product.discontinued?
```
Comment (damianlegawiec) [inline app/helpers/spree/admin/products_helper.rb:6]:
```suggestion
        return Spree.t(:archived) if product.status = 'archived'
```

File: app/helpers/spree/admin/products_helper.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        return Spree.t(:deleted) if product.deleted?
+
+        if product.available?
+          Spree.t(:active)
+        elsif product.available_on&.future?
+          Spree.t(:pending_sale)
```
Comment (damianlegawiec) [inline app/helpers/spree/admin/products_helper.rb:12]:
```suggestion
          Spree.t(:pre_order)
```

File: app/helpers/spree/admin/products_helper.rb (hunk 1 above)
Comment (kgorazd) [reply app/helpers/spree/admin/products_helper.rb]: done

File: app/helpers/spree/admin/products_helper.rb (hunk 2 above)
Comment (kgorazd) [reply app/helpers/spree/admin/products_helper.rb]: done

File: app/javascript/spree/dashboard/controllers/product_edit_controller.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  switchAvailabilityDatesFields(event) {
+    let status = event.target.value
+    if (status === "draft") {
+      this.show(this.availableOnTarget)
+      this.show(this.makeActiveAtTarget)
+      this.hide(this.discontinueOnTarget)
+    } else if (status === "active") {
+      this.hide(this.availableOnTarget)
```
Comment (damianlegawiec) [inline app/javascript/spree/dashboard/controllers/product_edit_controller.js:24]: @kgorazd available on should still be visible, this is the release date

File: app/javascript/spree/dashboard/controllers/product_edit_controller.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  connect() {
+    this.statusTarget.dispatchEvent(new Event("change"))
+  }
+
+  switchAvailabilityDatesFields(event) {
+    let status = event.target.value
+    if (status === "draft") {
+      this.show(this.availableOnTarget)
+      this.show(this.makeActiveAtTarget)
+      this.hide(this.discontinueOnTarget)
```
Comment (damianlegawiec) [inline app/javascript/spree/dashboard/controllers/product_edit_controller.js:22]: @kgorazd no need to hide this field

File: app/javascript/spree/dashboard/controllers/product_edit_controller.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    } else if (status === "active") {
+      this.hide(this.availableOnTarget)
+      this.hide(this.makeActiveAtTarget)
+      this.show(this.discontinueOnTarget)
+    } else {
+      this.hide(this.availableOnTarget)
+      this.hide(this.makeActiveAtTarget)
+      this.hide(this.discontinueOnTarget)
```
Comment (damianlegawiec) [inline app/javascript/spree/dashboard/controllers/product_edit_controller.js:30]: @kgorazd no need to hide this field in this case

File: app/javascript/spree/dashboard/controllers/product_edit_controller.js (hunk 3 above)
Comment (kgorazd) [reply app/javascript/spree/dashboard/controllers/product_edit_controller.js]: done

File: app/javascript/spree/dashboard/controllers/product_edit_controller.js (hunk 4 above)
Comment (kgorazd) [reply app/javascript/spree/dashboard/controllers/product_edit_controller.js]: done

File: app/javascript/spree/dashboard/controllers/product_edit_controller.js (hunk 5 above)
Comment (kgorazd) [reply app/javascript/spree/dashboard/controllers/product_edit_controller.js]: done

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**